node/src/components/linear_chain.rs

```diff
@@ -304,12 +303,9 @@ impl<I> LinearChain<I> {
             public_key,
         } = fs;
         debug!(%block_hash, %public_key, "removing finality signature from pending collection");
-        let validator_sigs = self
-            .pending_finality_signatures
-            .entry(*public_key)
-            .or_default();
-        validator_sigs.remove(&block_hash);
-        self.remove_empty_entries();
+        if let Some(validator_sigs) = self.pending_finality_signatures.get_mut(public_key) {
+            validator_sigs.remove(&block_hash);
+        }
```

Comment:
We should still `self.remove_empty_entries();`, though, or directly check whether the `public_key` entry is empty now.

Reply:
Ups :zipper_mouth_face: that was unintentional.

Reply:
Done in https://github.com/CasperLabs/casper-node/pull/773/commits/9bceba39b0f914af2c0d05cede2cfbf6b92430c9